server/handlers/changefeed.go

```diff
@@ -0,0 +1,79 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	ctxu "github.com/docker/distribution/context"
+	"github.com/gorilla/mux"
+	"golang.org/x/net/context"
+
+	"github.com/docker/notary"
+	"github.com/docker/notary/server/errors"
+	"github.com/docker/notary/server/storage"
+)
+
+type changefeedResponse struct {
+	NumberOfRecords int              `json:"count"`
+	Records         []storage.Change `json:"records"`
+}
+
+// Changefeed returns a list of changes according to the provided filters
+func Changefeed(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+	var (
+		vars                = mux.Vars(r)
+		logger              = ctxu.GetLogger(ctx)
+		qs                  = r.URL.Query()
+		imageName           = vars["imageName"]
+		changeID            = qs.Get("change_id")
+		store, records, err = checkChangefeedInputs(logger, ctx.Value(notary.CtxKeyMetaStore), qs.Get("records"))
+	)
+	if err != nil {
+		// err already logged and in correct format.
+		return err
+	}
+	out, err := changefeed(logger, store, imageName, changeID, records)
+	if err == nil {
+		w.Write(out)
+	}
+	return err
+}
+
+func changefeed(logger ctxu.Logger, store storage.MetaStore, imageName, changeID string, records int64) ([]byte, error) {
+	changes, err := store.GetChanges(changeID, int(records), imageName)
```

Comment:
The build is has been so flaky the last 48 hours I'm mildly afraid of making any more changes. I've filed issues to look at CI. Unless you feel really strongly about this I'm not going to change it with the tight timeline.